src/h5web/visualizations/line/LineVis.tsx

```diff
@@ -27,36 +33,62 @@ function LineVis(props: Props): ReactElement {
     curveType = CurveType.LineOnly,
     showGrid = true,
     scaleType = ScaleType.Linear,
+    abscissas,
   } = props;
 
-  const indexDomain = extendDomain([0, dataArray.size - 1], 0.01);
-  const dataDomain = useMemo(() => {
-    return domain
-      ? extendDomain(domain, 0.05, scaleType === ScaleType.Log)
+  const abscissaIndices = range(dataArray.size);
+  const abscissaValues = abscissas || abscissaIndices;
+  const abscissaToIndex = getValueToIndexScale(abscissaValues);
+  const abscissaDomain = useMemo(() => {
+    const rawDomain = getDataDomain(abscissaValues);
+    return rawDomain ? extendDomain(rawDomain, 0.01) : DEFAULT_DOMAIN;
+  }, [abscissaValues]);
+
+  const ordinateValues = dataArray.data as number[];
+  const ordinateDomain = useMemo(() => {
+    const rawDomain = getDataDomain(ordinateValues);
+    return rawDomain
+      ? extendDomain(rawDomain, 0.05, scaleType === ScaleType.Log)
       : DEFAULT_DOMAIN;
-  }, [scaleType, domain]);
+  }, [ordinateValues, scaleType]);
+
+  if (abscissaValues.length !== dataArray.size) {
+    return (
+      <p>{`Abscissas size (${abscissaValues.length}) does not match data length (${dataArray.size})`}</p>
+    );
+  }
 
   return (
     <div className={styles.root}>
       <VisCanvas
-        abscissaConfig={{ indexDomain, showGrid }}
-        ordinateConfig={{ dataDomain, showGrid, scaleType }}
+        abscissaConfig={{
+          domain: abscissaDomain,
+          showGrid,
+          isIndexAxis: !abscissas,
+        }}
+        ordinateConfig={{
+          domain: domain || ordinateDomain,
```

Comment:
Note that `LineVis` still accepts a `domain` prop that will take precedence over the `ordinateDomain`. This is useful for the `autoScale` behaviour where the `domain` is inferred from the whole dataset rather than from the slice.

Reply:
Is this change required for now? I'm not sure I agree with it, so I'd like to discuss it further. Note that, as is, it changes the behaviour, since the domain that is passed when auto-scale is off is no longer extended.

Reply:
The change is not **required** per se but not making it breaks the Autoscale feature as the domain will be always inferred from the data passed to the LineVis.

Reply:
In fact, this is a fix to an unneeded change I made that puts the computation of the domain in the LineVis. I will revert all of this.